### PR TITLE
Set buildpack version as header on dcl upload

### PR DESCRIPTION
**Pull request by f-blass**
better traceability which versions are used.

### CODE REVIEW

**Review by d047491 — COMMENTED** [review]
I don't fully understand the approach here. There is a setter, that is always called und gets the value from the manifest. It does not look clean to me. I see better ways to achieve that. I did not think it trough yet but. I have two ideas:

1. Let the uploader read the version from the manifest directly
2. Or better: parameterize the User-Agent Header value for the uploader (for example add it to the Uploader struct type)

**Review by d047491 — CHANGES_REQUESTED** [review]
reason: see last comment

**Comment by f-blass** [comment]
Originally I wanted to use build flags, which however is not possible with the buildpack packager. 

I thought about adding it to the Uploader struct aswell. I didn't like it either, because you would need to pass it also to all other structs if required somewhere else. Adding the whole manifest to the uploader seems even worse, because the uploader should not even need to know that it is running in the context of a buildpack.

Since the project is rather small and we do not have HTTP calls anywhere else outside of the uploader, I'll add it to the Uploader struct for now.

File: pkg/supply/supply.go [hunk 1]
```diff
@@ -284,6 +285,7 @@ func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, root
 		Root:          path.Join(s.Stager.BuildDir(), rootDir),
 		Client:        client,
 		AMSInstanceID: creds.AmsInstanceID,
+		UserAgent:     fmt.Sprintf("cloud-authorization-buildpack/%s", buildpackVersion),
```
Comment (d047491) [inline pkg/supply/supply.go:288]:
```suggestion
		UserAgent:     fmt.Sprintf("cloud-authorization-buildpack/%s",  s.BuildpackVersion),
```

Comment (d047491) [reply pkg/supply/supply.go:288]: no need to make to provide an attribute as a parameter to a private funtion.

File: pkg/supply/supply.go [hunk 2]
```diff
@@ -274,7 +275,7 @@ func (s *Supplier) supplyCertCopier() error {
 	return os.Chmod(destFile, 0755)
 }
 
-func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, rootDir string) error {
+func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, rootDir, buildpackVersion string) error {
```
Comment (d047491) [inline pkg/supply/supply.go:278]:
```suggestion
func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, rootDir) error {
```

File: pkg/supply/supply.go [hunk 3]
```diff
@@ -101,7 +102,7 @@ func (s *Supplier) Run() error {
 		return fmt.Errorf("could not write profileD file: %w", err)
 	}
 	if cfg.ShouldUpload {
-		if err := s.upload(identityCreds, tlsCfg, cfg.Root); err != nil {
+		if err := s.upload(identityCreds, tlsCfg, cfg.Root, s.BuildpackVersion); err != nil {
```
Comment (d047491) [inline pkg/supply/supply.go:105]:
```suggestion
		if err := s.upload(identityCreds, tlsCfg, cfg.Root); err != nil {
```

**Review by d047491 — CHANGES_REQUESTED** [review]
just a little beautification.

**Review by d047491 — APPROVED** [review]
looks good now :-)

**Pull request merged**